src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -800,8 +800,10 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
             var htmlSummary = new TagBuilder("ul");
             foreach (var modelState in modelStates)
             {
-                foreach (var modelError in modelState.Errors)
+                // Perf: Avoid allocations
+                for (var i = 0; i < modelState.Errors.Count; i++)
```

Comment:
It's inside a for-loop. That's one enumerator allocation per modelstate entry. This was allocating 3 enumerators per request for a very simple site.